dbms/src/Flash/tests/gtest_compute_server.cpp

```diff
@@ -764,6 +794,56 @@ try
 }
 CATCH
 
+TEST_F(ComputeServerRunner, testErrorMessage)
+try
+{
+    startServers(3);
+    setCancelTest();
+    std::vector<String> failpoint_names{
+        FailPoints::exception_before_mpp_register_non_root_mpp_task,
+        FailPoints::exception_before_mpp_register_root_mpp_task,
+        FailPoints::exception_before_mpp_register_tunnel_for_non_root_mpp_task,
+        FailPoints::exception_before_mpp_register_tunnel_for_root_mpp_task,
+        FailPoints::exception_during_mpp_register_tunnel_for_non_root_mpp_task,
+        FailPoints::exception_before_mpp_non_root_task_run,
+        FailPoints::exception_before_mpp_root_task_run,
+        FailPoints::exception_during_mpp_non_root_task_run,
+        FailPoints::exception_during_mpp_root_task_run,
+    };
+    size_t query_index = 0;
+    for (const auto & failpoint : failpoint_names)
+    {
+        query_index++;
+        for (size_t i = 0; i < 5; i++)
+        {
+            auto properties = DB::tests::getDAGPropertiesForTest(serverNum(), query_index, i);
+            MPPQueryId query_id(properties.query_ts, properties.local_query_id, properties.server_id, properties.start_ts);
+            FailPointHelper::enableFailPoint(failpoint);
```

Comment:
Does failpoint need to be turned off at L841?

Reply:
Not for now because all the failpoint is automatically disabled after triggered once, but I think it's ok to disable it manually.